react/features/video-quality/middleware.js

```diff
@@ -46,21 +47,46 @@ StateListenerRegistry.register(
         const { reducedUI } = state['features/base/responsive-ui'];
         const _shouldDisplayTileView = shouldDisplayTileView(state);
         const thumbnailSize = state['features/filmstrip']?.tileViewDimensions?.thumbnailSize;
+        const participantCount = getParticipantCount(state);
 
         return {
             displayTileView: _shouldDisplayTileView,
+            participantCount,
             reducedUI,
             thumbnailHeight: thumbnailSize?.height
         };
     },
-    /* listener */ ({ displayTileView, reducedUI, thumbnailHeight }, { dispatch, getState }) => {
-        const { maxReceiverVideoQuality } = getState()['features/base/conference'];
+    /* listener */ ({ displayTileView, participantCount, reducedUI, thumbnailHeight }, { dispatch, getState }) => {
+        const state = getState();
+        const { maxReceiverVideoQuality } = state['features/base/conference'];
+        let { maxFullResolutionParticipants } = state['features/base/config'];
+
+        if (typeof maxFullResolutionParticipants === 'undefined') {
+            maxFullResolutionParticipants = 4;
```

Comment:
You can avoid this by doing: `const { maxFullResolutionParticipants = 4 } = state['features/base/config'];`